code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -105,12 +104,23 @@
 				M.satiety -= junkiness
 			playsound(M.loc,'sound/items/eatfood.ogg', rand(10,50), 1)
 			if(reagents.total_volume)
-				var/fraction = min(bitesize/reagents.total_volume, 1)
+				var/fraction = min(bitesize / reagents.total_volume, 1)
 				reagents.reaction(M, INGEST, fraction)
 				reagents.trans_to(M, bitesize)
 				bitecount++
 				On_Consume()
-			return 1
+				if(iscarbon(M))
```

Comment:
I'm not sure how you'd like me to modularize this. Any pointers on what I could change maybe? I'll try get it done once I know what I should change.